lambda/send-event-to-txma/send-event-to-txma.ts

```diff
@@ -0,0 +1,122 @@
+import { SNSEvent } from "aws-lambda";
+import {
+  SendMessageCommand,
+  SendMessageCommandOutput,
+  SendMessageRequest,
+  SQSClient,
+} from "@aws-sdk/client-sqs";
+import { TxmaEvent } from "./models";
+import {
+  COMPONENT_ID,
+  EventNamesEnum,
+  REPORT_SUSPICIOUS_ACTIVITY_EVENT_NAME,
+  ValidationRulesKeyEnum,
+} from "./constants";
+import VALIDATOR_RULES_MAP from "./validator-rules";
+import validateObject from "./validator";
+
+export const transformToTxMAEvent = (event: any, eventName: string): any => {
+  let txmaEvent = null;
+  if (eventName === EventNamesEnum.HOME_REPORT_SUSPICIOUS_ACTIVITY) {
+    txmaEvent = {
+      component_id: COMPONENT_ID,
+      event_name: REPORT_SUSPICIOUS_ACTIVITY_EVENT_NAME,
+      user: {
+        user_id: event.user_id,
+        persistent_session_id: event.persistent_session_id,
+        session_id: event.session_id,
+      },
+      extensions: {
+        reported_session_id: event.reported_event.session_id,
+      },
+    };
+  } else {
+    throw new Error(
+      "Unsupported event - There is no transformation logic for this event"
+    );
+  }
+  return txmaEvent;
+};
+
+export const sendSqsMessage = async (
+  messageBody: string,
+  queueUrl: string | undefined
+): Promise<SendMessageCommandOutput> => {
+  const { AWS_REGION } = process.env;
+  const client = new SQSClient({ region: AWS_REGION });
+  const message: SendMessageRequest = {
+    QueueUrl: queueUrl,
+    MessageBody: messageBody,
+  };
+  return client.send(new SendMessageCommand(message));
+};
+/**
+ * A function for sending an audit event message to the TxMA SQS queue.
+ *
+ * @param txmaEvent - the transformed txma event
+ * @param queueUrl
+ * @returns the response returned by the SQS client
+ * @throws error
+ * Thrown when an exception is thrown while trying to send a message to the queue
+ */
+export async function sendAuditEvent(
+  txmaEvent: TxmaEvent,
+  queueUrl: string | undefined
+): Promise<SendMessageCommandOutput> {
+  try {
+    return await sendSqsMessage(JSON.stringify(txmaEvent), queueUrl);
+  } catch (error: any) {
+    console.error(
+      `Error occurred trying to send the audit event to the TxMA queue: ${error.message}`
+    );
+    throw error;
+  }
+}
+
+export const handler = async (event: SNSEvent): Promise<void> => {
+  const { DLQ_URL, EVENT_NAME, TXMA_QUEUE_URL } = process.env;
+  await Promise.all(
+    event.Records.map(async (record) => {
+      try {
+        if (EVENT_NAME) {
```

Comment:
Consider reversing the logic of this statement and turning it into a guard condition check.